site-src/guides/migrating-from-ingress.md

```diff
@@ -0,0 +1,261 @@
+
+The table below summarizes the mapping between the Ingress API and the Gateway API personas:
+
+| Ingress API Persona | Gateway API Persona |
+|-|-|
+| User | Application developer, Application admin, Cluster operator |
+| Cluster operator | Cluster operator |
+| Infrastructure provider | Infrastructure provider | 
+
+### Available Features 
+
+The Ingress API comes with basic features only: TLS termination and content-based routing of HTTP traffic based on the host header and the URI of a request. To offer more features, Ingress controllers support them through annotations on the Ingress resource which are vendor-specific extensions to the Ingress API.
+
+The annotations approach to extensibility has two negative consequences for users of the Ingress API: 
+
+* *Limited portability*. Because so many features are available through annotations, switching between Ingress controllers becomes difficult or even impossible, as it is necessary to convert the annotations of one vendor to another (the other vendor might not even support some features of the first one). This limits the portability of the Ingress API.
+* *Awkwardness of the API*. Because annotations are key-value strings (as opposed to a structured scheme like the spec of the Ingress resource) and applied at the top of a resource (rather than in the relevant parts of the spec), the Ingress API can become awkward to use, especially when a large number of annotations are added to an Ingress resource.
+
+The Gateway API supports all the features of the Ingress resources and many features that are only available through annotations. As a result, the Gateway API is more portable than the Ingress API. Additionally, as the next section will show, you will not need to use any annotations at all, which addresses the awkwardness problem. 
+
+### Approach to Extensibility
+
+The Ingress API has two extensions points:
+
+* Annotations on the Ingress resource (described in the previous section).
+* [Resource backends](https://kubernetes.io/docs/concepts/services-networking/ingress/#resource-backend), which is the ability to specify a backend other than a Service.
+
+The Gateway API is feature-rich compared with the Ingress API. However, to configure some advanced features like authentication or common but non-portable across data planes features like connection timeouts and health checks, you will need to rely on the extensions of the Gateway API.
+
+The Gateway API has the following primary extension points:
+
+* *External references.* A feature (field) of a Gateway API resource can reference a custom resource (CRD) specific to the Gateway implementation that configures that feature. For example:
```

Comment:
Tiny nit, I'd either remove "CRD" or replace it with "CR" to more accurately represent the "custom resource" terminology:

```suggestion
* *External references.* A feature (field) of a Gateway API resource can reference a custom resource specific to the Gateway implementation that configures that feature. For example:
```